### PR TITLE
fix indenting

### PR DESCRIPTION
**Pull request by cehsu**
This change is [Reviewable](https://reviewable.io/reviews/sfbrigade/sfsbook/94)

### CODE REVIEW

**Comment by rjkroege** [comment]
Reviewed 11 of 19 files at r1.
Review status: 11 of 19 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[site/head.html, line 7 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZReeA17HUkD5Pcr1Te:-KZReeA17HUkD5Pcr1Tf:b2vcrz5) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/head.html#L7)):*
> ```HTML
> 	<script type="text/javascript" src="/js/example.js"></script>
> 	<script type="text/javascript" src="/js/searchbar.js"></script>
> 	<link href="https://fonts.googleapis.com/css?family=Source+Sans+Pro" rel="stylesheet">
> ```

web fonts are frequently a performance problem on slow networks. But beautiful fonts are extremely important to the overall impression of the site. I don't know how we should trade that off. Let's discuss at a January hack-night.

---

*[site/login.html, line 24 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRgdfwK-thynUgo4P7:-KZRgdfwK-thynUgo4P8:b-kdwa2i) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/login.html#L24)):*
> ```HTML
> 			<h3>Welcome</h3>
> 			<div class="onefield">
> 				<input class="small-input login" placeholder="Username" id="username" type="text" name="username" size="30" ./>
> ```

The Mozilla docs say that one should not use the `placeholder` attribute in place of a `label` argument but it looks like that's what you've done here? Are the moz docs wrong? So what's your reasoning for removing the label?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94)*

**Comment by rjkroege** [comment]
Reviewed 7 of 19 files at r1.
Review status: 18 of 19 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[site/js/searchbar.js, line 11 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRhL2kfPtaIeBafslf:-KZRhL2kfPtaIeBafslg:bn7o2lh) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/js/searchbar.js#L11)):*
> ```JavaScript
> 	var stringToReplace = optionValue.concat(', ');
> 
> 	// CEH: need to determine whether user wants to be able to toggle selection via dropdown
> ```

Google style is 80 character lines

---

*[site/js/searchbar.js, line 13 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRhXFkWO2q8whoPl5M:-KZRhXFkWO2q8whoPl5N:b-s36g8q) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/js/searchbar.js#L13)):*
> ```JavaScript
> 	// CEH: need to determine whether user wants to be able to toggle selection via dropdown
> 	if(searchText.match(stringToReplace)) {
> 		//stringToReplace = optionValue.concat(', ');
> ```

this is all commented out? If we don't need it, you could remove?

---

*[site/js/searchbar.js, line 29 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRhix4wocsZW-prwXt:-KZRhix4wocsZW-prwXu:bmdjz95) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/js/searchbar.js#L29)):*
> ```JavaScript
> 			searchfield.value=searchfield.value.replace(buttonValue, '');
> 		};
> 		btn.onclick = function(event){
> ```

space before {

---

*[site/js/searchbar.js, line 36 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRhpgf3AgXZkbPstqK:-KZRhpgf3AgXZkbPstqL:b-7ropln) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/js/searchbar.js#L36)):*
> ```JavaScript
> 		}
> 		hbox.appendChild(btn);
> 		if(searchText.length > 0){
> ```

ditto

---

*[site/js/searchbar.js, line 60 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRhyR_N9ETuvIiSZdM:-KZRhyRanYScugc_E_EX:b-ibjjmm) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/js/searchbar.js#L60)):*
> ```JavaScript
> 	var navbar = document.getElementsByClassName('nav')[0];
> 	console.log('toggleactivenav firing', navbar);
>   if(navbar.classList.length > 1){
> ```

inconsistent space

---

*[site/js/searchbar.js, line 94 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRi10AcRRJHWsEjOaI:-KZRi10AcRRJHWsEjOaJ:bxq5dco) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/js/searchbar.js#L94)):*
> ```JavaScript
> // attachToggles attaches event listener to each category option
> function attachToggles() {
> 	//CEH: dry this up
> ```

what does this mean? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94)*

**Comment by rjkroege** [comment]
I love how the use of multi-templates has simplified the code. Awesome! And: btw, the test failure is real. Something needs to be updated.

**Comment by cehsu** [comment]
Review status: 16 of 19 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[site/login.html, line 24 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRgdfwK-thynUgo4P7:-KZSrETxNe5EA4jsT0Ul:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/login.html#L24)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

The Mozilla docs say that one should not use the `placeholder` attribute in place of a `label` argument but it looks like that's what you've done here? Are the moz docs wrong? So what's your reasoning for removing the label?
</blockquote></details>

Done.

---

*[site/js/searchbar.js, line 11 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRhL2kfPtaIeBafslf:-KZSrGq0OOTdfHJPUHyq:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/js/searchbar.js#L11)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

Google style is 80 character lines
</blockquote></details>

Done.

---

*[site/js/searchbar.js, line 13 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRhXFkWO2q8whoPl5M:-KZSrHtOs6wAskAR2-AR:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/js/searchbar.js#L13)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

this is all commented out? If we don't need it, you could remove?
</blockquote></details>

Done.

---

*[site/js/searchbar.js, line 29 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRhix4wocsZW-prwXt:-KZSrIbEVfxvxf4soEHS:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/js/searchbar.js#L29)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

space before {
</blockquote></details>

Done.

---

*[site/js/searchbar.js, line 36 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRhpgf3AgXZkbPstqK:-KZSrJ44ZIpPw_ahGxuC:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/js/searchbar.js#L36)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

---

*[site/js/searchbar.js, line 60 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRhyR_N9ETuvIiSZdM:-KZSrJnWauPrsdi90mAK:b-896fix) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/js/searchbar.js#L60)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

inconsistent space
</blockquote></details>

Done.

---

*[site/js/searchbar.js, line 94 at r1](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-KZRi10AcRRJHWsEjOaI:-KZSrOKz3RHGqn74j1dq:b-yf136) ([raw file](https://github.com/sfbrigade/sfsbook/blob/4742252b141eb4a78366edc0ec783ec888ee3cf2/site/js/searchbar.js#L94)):*
<details><summary><i>Previously, rjkroege (Robert Kroeger) wrote…</i></summary><blockquote>

what does this mean? 
</blockquote></details>

a reminder to "do not repeat"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94)*

**Comment by rjkroege** [comment]
Reviewed 3 of 4 files at r2.
Review status: 18 of 19 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94)*

**Comment by rjkroege** [comment]
Reviewed 1 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94)*

**Comment by rjkroege** [comment]
:lgtm:

But you need to get the tests to pass. :-)

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94#-:-KZUMH4w_L6Uq5eYkkEj:b-5pjn72)*

**Comment by rjkroege** [comment]
The tests are currently failing because this change updates a number of tempting sites with an extended list of subordinate templates but doesn't update the mock used in the test as well. For example,

```go
	templates := []string{sn, "/head.html", "/header.html", "/searchbar.html", "/footer.html"}
```
line in `usermgt.go` needs additional mocking entries for the mock resource entry. Something like this:

```go
func testHelper() func() {
	stashedResources := Resources

	Resources = map[string]string{
		"/usermgt/changepasswd.html": embeddedResourceForUsermgt,
		"/head.html": "",
		"/header.html": "",
		"/searchbar.html": "",
		"/footer.html": "",
	}

	return func() { Resources = stashedResources }
}
```

in `usermgt_test.go` fixes multiple tests.

File: server/usermgt_test.go [hunk 1]
```diff
@@ -28,6 +28,10 @@ func testHelper() func() {
 
 	Resources = map[string]string{
 		"/usermgt/changepasswd.html": embeddedResourceForUsermgt,
+    "/head.html": "",
```
Comment (rjkroege) [inline server/usermgt_test.go]: This isn't indented correctly. You can fix this with `gofmt`

File: site/css/style.css [hunk 2]
```diff
@@ -1,50 +1,214 @@
...
+  vertical-align: super;
+}
+th {
+  text-align: left;
+}
+.user-details th, .user-details td {
+  padding: 5px;
+  text-align: right;
+}
 a {
```
Comment (rjkroege) [inline site/css/style.css]: You would seem to have defined `a` twice?

File: site/login.html [hunk 3]
```diff
@@ -1,55 +1,45 @@
...
-
-{{ template "footer" . }}
-
-</body>
+	{{ template "head" . }}
+	<body>
+		{{ template "header" . }}
+		<!-- need to add a status bar -->
+
+    {{ template "searchbar" . }}
```
Comment (rjkroege) [inline site/login.html]: The indenting is weird?

File: site/resources/resource.html [hunk 4]
```diff
@@ -1,151 +1,161 @@
 <html>
-<head>
-<script type="text/javascript" src="/js/example.js"></script>
-<link href="/css/style.css" rel="stylesheet">
-</head>
-<body>
+	{{ template "head" . }}
+	<body>
 
-{{ template "header" . }}
```
Comment (rjkroege) [inline site/resources/resource.html]: More strange indenting?

File: server/usermgt_test.go (hunk 1 above)
Comment (cehsu) [inline server/usermgt_test.go]: good point, resolved in test cl.

File: site/css/style.css (hunk 2 above)
Comment (cehsu) [inline site/css/style.css]: it looks like there was a redundant style assignment (color) but it makes sense to distinguish two different rules

File: site/login.html (hunk 3 above)
Comment (cehsu) [inline site/login.html]: good point, resolved in test cl.

File: site/resources/resource.html (hunk 4 above)
Comment (cehsu) [inline site/resources/resource.html]: good point, resolved in test cl.

**Comment by rjkroege** [comment]
Reviewed 2 of 3 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/sfbrigade/sfsbook/94)*

**Comment by rjkroege** [comment]
This PR is obsolete? Can you drop it in the interests of neatness?

**Pull request closed without merging**